src/Repositories/MemoryRepository.php

```diff
@@ -26,27 +26,20 @@ class MemoryRepository extends BaseRepository implements Repository
     /** @var int[] */
     private $reservedMessageIds = [];
 
-    /** @var SplObjectStorage|TopicSubscription[] */
-    private $topicSubscriptions;
+    /** @var array<int, TopicSubscription> */
+    private $subscriptions;
 
-    /** @var SplObjectStorage|PublishedMessage[] */
-    private $pendingPublishedMessages;
+    /** @var SplQueue */
+    private $subscriptionsQueue;
 
-    /** @var SplObjectStorage|UnsubscribeRequest[] */
-    private $pendingUnsubscribeRequests;
-
-    /** @var SplObjectStorage|PublishedMessage[] */
-    private $pendingPublishConfirmations;
+    /** @var array<int, PublishedMessage> */
+    private $pendingMessages;
```

Comment:
Shouldn't arrays be initialized at this point?

Reply:
And would it make sense to use a queue for pending messages, so you only need to pop items until you find one which has a `$lastSentAt` date time below the threshold? Popped items could then be thrown onto the queue again.

Reply:
The queue is a no go since the specs don't mandate that multiple messages must be delivered for overlapping subscriptions.. so will be gone tomorrow